src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1398,10 +1386,15 @@ export default class DefaultDeviceController
       this.logger.info(`reusing existing ${kind} input device`);
       return;
     }
-
-    if (kind === 'audio' && this.activeDevices[kind] && this.activeDevices[kind].stream) {
-      this.releaseActiveDevice(this.activeDevices[kind]);
+    if (this.activeDevices[kind] && this.activeDevices[kind].stream) {
+      /* istanbul ignore else */
+      if (kind === 'audio') {
+        this.releaseActiveDevice(this.activeDevices[kind]);
+      } else if (kind === 'video') {
+        this.stopTracksAndRemoveCallback('video');
```

Comment:
So we stop tracks and release it later after getUserMedia? is it just to be safe or do you know the reason why can't we just call `releaseActiveDevice` here?

Reply:
The reason we cant call releaseActiveDevice is because it calls stopLocalVideo tile. The problem with that is that in `restartLocalVideoAfterSelection` we only `restartLocalVideo` if `hasStartedLocalVideoTile` is true. 

Meaning we wont actually update the peerConnection if stop the localVideoTile.